### PR TITLE
#165020179 | #165020180 Bookmark an article and list all my bookmarks

### PR DESCRIPTION
**Pull request by niyobobo**
### What does this PR do?
User should be able to bookmark an article and see list of his/her bookmarks
### Description of Task to be completed?
- Authenticated user will be able to  bookmark an article
- Authenticated user will be able to see the list of his bookmarks 
### How should this be manually tested?
- Clone the repository locally with `git clone https://github.com/andela/ah-kgl-avengers-frontend.git` command.
- Check out to this branch by running `git checkout ft-bookmark-article-165020179`.
- Install dependencies by running `npm install`.
- Update the .env file according to the .env.sample file
- Start the server with `npm start` and put this url `http://localhost:8000/` in the address bar.
- Bookmark an article by clicking on the bookmark icon shown in the article. This article will be added to your bookmark list
- Check your bookmark by clicking on an avatar or your image profile shown on the top-right corner, from the dropdown menus, my bookmarks then the new page with your bookmark list will be displayed.
### Any background context you want to provide?
- User should have an account on `AuthorsHaven` to be able to bookmark an article.
### What are the relevant pivotal tracker stories?
[#165020179](https://www.pivotaltracker.com/story/show/165020178)
[#165020180](https://www.pivotaltracker.com/story/show/165020180)
### Screenshots (if appropriate)
image

### Questions:
N/A

### CODE REVIEW

File: src/components/container/bookmark.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    const { history } = this.props;
+    if (!localStorage.token) history.push('/');
+    const { allBookmarks } = this.props;
+    allBookmarks();
+  }
+
+  deleteBookmark = async (slug) => {
+    const { remove } = this.props;
+    await remove(slug);
+    const { toastManager, bookmark } = this.props;
```
Comment (fridolinho) [inline src/components/container/bookmark.jsx]: Hey @niyobobo, your codes are working well but you are destructuring this.props in 2 lines 19 and 21, please do it in once for all those variables.

Comment (niyobobo) [reply src/components/container/bookmark.jsx]: Thank you @fridolinho for the comment, but the logic behind is this is that when a response comes to confirm that your response removed, it goes directly to the initial state that's why I need to destructure it and see what is there.

Comment (fridolinho) [reply src/components/container/bookmark.jsx]: Okay Got it.

**Review by fridolinho — APPROVED** [review]

**Review by aaronsekisambu — APPROVED** [review]
This works as we discussed on the call. Awesome word @niyobobo

File: src/app.jsx [hunk 2]
```diff
@@ -33,6 +34,7 @@ const App = () => (
       <Route exact path="/new-post" component={Editor} />
       <Route exact path="/my-articles" component={Articles} />
       <Route exact path="/reset" component={Reset} />
+      <Route exact path="/me/bookmarks" component={Bookmark} />
```
Comment (ezkemboi) [inline src/app.jsx:37]: Better work generally, but we could have had a better URL than using `/me`

File: src/components/container/Welcome.jsx [hunk 3]
```diff
@@ -19,8 +20,21 @@ class Welcome extends Component {
     onFetchArticles();
   }
 
+  bookmarkArticle = async (slug) => {
+    const { bookmarking } = this.props;
+    await bookmarking(slug);
+    const { bookmark } = this.props;
+    if (bookmark) {
+      if (bookmark.includes('jwt')) {
```
Comment (ezkemboi) [inline src/components/container/Welcome.jsx:28]: Good work adding toast when the user is not logged in.

File: src/components/functional/navBar.jsx [hunk 4]
```diff
@@ -96,7 +95,7 @@ class AppBar extends Component {
                     <Link to="/my-articles">
                       <DropdownItem>My Articles</DropdownItem>
                     </Link>
-                    <Link to="/">
+                    <Link to="/me/bookmarks">
```
Comment (ezkemboi) [inline src/components/functional/navBar.jsx:98]: As mentioned earlier on having better names than `/me`

**Review by ezkemboi — APPROVED** [review]
Good work generally.

**Pull request merged**